### PR TITLE
Misc coverity fixes

### PR DESCRIPTION
**Pull request by xrmx**
Mostly missing null termination for strings and an off by one

### CODE REVIEW

File: core/mule.c [hunk 1]
```diff
@@ -451,7 +451,7 @@ void uwsgi_setup_mules_and_farms() {
 			mules_list++;
 
 			strncpy(uwsgi.farms[i].name, farm_value, 0xff);
-			
+			uwsgi.farms[i].name[0xff-1] = '\0';
```
Comment (akx) [inline core/mule.c]: `0xff-1` seems like an awfully convoluted way to write 254...

Comment (xrmx) [inline core/mule.c]: Agree that we can improve by substituting 0xff magic number with a definition.

**Comment by xrmx** [comment]
@unbit is this good to go right? just rebased so it apply fine.

**Comment by Darvame** [comment]
Sorry, for slow reaction, but
```c
strncpy(use->receiver, who, len);
```
lua provides len without terminator, so we need do
```c
strncpy(use->receiver, who, len + 1); //also copy terminator from 'who'
```
or
```
strcpy(use->receiver, who); //len check has been passed, 'who' has terminator
```

File: plugins/lua/lua_plugin.c [hunk 2]
```diff
@@ -983,7 +983,7 @@ static int uwsgi_api_register_signal(lua_State *L) {
 		uwsgi_lock(uwsgi.signal_table_lock);
 
 		// master-proc checks receiver
-		strcpy(use->receiver, who);
+		strncpy(use->receiver, who, len);
```
Comment (Darvame) [inline plugins/lua/lua_plugin.c]: Sorry, for slow reaction, but
```c
strncpy(use->receiver, who, len);
```
lua provides len without terminator, so we need do
```c
strncpy(use->receiver, who, len + 1); //also copy terminator from 'who'
```
or
```
strcpy(use->receiver, who); //len check has been passed, 'who' has terminator
```

Comment (xrmx) [inline plugins/lua/lua_plugin.c]: Il 07/12/2015 19:50, Darvame Hleran ha scritto:
> In plugins/lua/lua_plugin.c
> <https://github.com/unbit/uwsgi/pull/1084#discussion_r46859422>:
>
>> @@ -983,7 +983,7 @@ static int uwsgi_api_register_signal(lua_State *L) {
>>  		uwsgi_lock(uwsgi.signal_table_lock);
>>
>>  		// master-proc checks receiver
>> -		strcpy(use->receiver, who);
>> +		strncpy(use->receiver, who, len);
>
> Sorry, for slow reaction, but
>
> strncpy(use->receiver, who, len);
>
> lua provides len without terminator, so we need do
>
> strncpy(use->receiver, who, len +1);//also copy terminator from 'who'
>
> or
>
> |strcpy(use->receiver, who); //len check has been passed, 'who' has terminator |

Thanks fixed. Forgot to give you proper credit, sorry.


-- 
Riccardo Magliocchetti
@rmistaken

http://menodizero.it

**Pull request merged**